server.js

```diff
@@ -18,16 +19,58 @@ import cors from 'cors'
 const port = process.env.PORT || 8080
 const app = express()
 
+
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
+
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+app.get('/', (request, response) => {
+  response.send('Hello world');
+});
+
+
+app.get('/books', (request, response) => {
+  response.json(booksData);
+});
+
+
+app.get('/books/book/:bookID', (request, response) => {
+  const bookID = request.params.bookID;
+  const book = booksData.find((item) => item.bookID === +bookID);
+
+  if (!book) {
+    response.send('Could not find this book due to incorrect ID!')
+  };
+  
+  response.json(book);
+});
+
+
+app.get('/rating/:rating', (request, response) => {
+  const choosenRating = request.params.rating;
+  let ratedBooks = booksData.filter((book) => book.average_rating <= +choosenRating);
+
+  response.json(ratedBooks);
+});
```

Comment:
Endpoints should be named after what they are returning, not after how they are filtered. This could therefore be a query param under your main endpoint, something like:
`/books?rating=descending`